experimental/addons/hive-operator/README.md

```diff
@@ -0,0 +1,142 @@
+# hive-operator
+
+Apache Hive is a distributed, fault-tolerant data warehouse system that enables analytics at a massive scale. Hive Metastore(HMS) provides a central repository of metadata that can easily be analyzed to make informed, data driven decisions, and therefore it is a critical component of many data lake architectures. Hive is built on top of Apache Hadoop and supports storage on S3, adls, gs etc though hdfs. Hive allows users to read, write, and manage petabytes of data using SQL.
+
+This is an operator for Kubernetes that can manage Apache Hive. Currently, it only supports the Hive Metastore!
+
+## Install Operator
+
+Add experimental registry
+```
+vela addon registry add experimental --type=helm --endpoint=https://addons.kubevela.net/experimental/
+```
+
+Enable this addon
+```
+vela addon enable hive-operator
+```
+
+```shell
+$ vela ls -A | grep hive
+vela-system     addon-hive-operator     ns-hive-operator                        k8s-objects                                     running healthy
+vela-system     └─                      hive-operator                           helm                                            running healthy Fetch repository successfully, Create helm release
+```
+
+Disable this addon
+```
+vela addon disable hive-operator
+```
+
+## Use
+
+After going through the Installation section and having installed all the dependencies, you will now deploy a Hive metastore cluster and it’s dependencies. Afterwards you can verify that it works.
+
+In order to connect Hive to MinIO we need to create several dependent components like S3Connection, Secret and a SecretClass
+
+**s3-connection**
+
+An S3Connection to connect to MinIO, Apply below YAML:
+
+```yaml
+apiVersion: core.oam.dev/v1beta1
+kind: Application
+metadata:
+  name: s3-connection-sample
+spec:
+  components:
+    - type: s3-connection
+      name: minio
+      properties:
+        host: minio
+        port: 9000
+        accessStyle: Path
+        credentials:
+          secretClass: hive-s3-secret-class
```

Comment:
Why not put all these (including the single-component Applications below) into one single Application that the user can run and test? If used in this way, it is essentially the same as applying a bunch of yamls.